### DISCUSSION TITLE
Pipeline operator 目前有啥进展吗？

### DISCUSSION THREAD

**Discussion by xiaoxiangmoe** [JSCIG/es-discuss#1] [Q&A] [answered]
已经卡在 stage 1 好久了。需要如何推进？

**Comment by hax** [comment, marked as answer]
最主要的问题，要再推进之前得决定到底走什么方向。因为pipeline有几个并行的提案，目前还留存的提案是 F# style （再加上后续 partial application 提案）和 topic style（smart style）。但其实 Elixir style 或者原本 bind operator 的形式也各有千秋。

我个人感觉，强行要在这几个形式中选择，比较大可能性是上 F# style，但结果也不见得好。我在准备一个新提案 extensions（形式上比较接近原本的 bind operator），基于这个提案，pipe（和 partial application）的需求可以通过 userland 库来解决（后续也可以对库标准化）。见 #3 。
